scarb/src/ops/resolve.rs

```diff
@@ -390,6 +391,61 @@ fn cairo_compilation_unit_for_target(
         member.id
     };
 
+    // Collect dependencies for the components.
```

Comment:
This logic is quite indirect in this context, because you have access to `WorkspaceResolve`, which keeps a graph of all packages in CU.

Reply:
Discussed offline - will be changed later by Scarb team